.ci/build-app-image.sh

```diff
@@ -4,13 +4,17 @@ set -e 	# Exit immediately upon failure
 : ${1?"Need to pass BASE_IMAGE as argument"}
 : ${2?"Need to pass TEST_APP as argument"}
 : ${3?"Need to pass TAG as argument"}
+: ${4?"Need to pass BRANCH as argument"}
 
 BASE_IMAGE=$1
 TEST_APP=$2
 TAG=$3
+BRANCH=$4
 
 echo "[CI] Injecting Dockerfile to project $TEST_APP..."
 cd  $SAMPLES_REPO/samples/$TEST_APP
+echo "[CI] Checkout branch $BRANCH..."
+git checkout $BRANCH
```

Comment:
could you also add  `git checkout -` after this command finishes? That could make sure we switch back to the old branch if we are done with this script successfully.